movement/watch_faces/complication/totp_face_lfs.c

```diff
@@ -279,15 +286,37 @@ static void totp_face_display(totp_lfs_state_t *totp_state) {
     watch_display_string(buf, 0);
 }
 
+static void totp_face_lfs_switch_to_secmod(totp_lfs_state_t *totp_state) {
+    int i;
+
+    for (i = 0; i < num_totp_records; ++i) {
+        totp_face_lfs_get_file_secret(&totp_records[i]);
+        totp_records[i].secret = malloc(totp_records[i].secret_size);
+        if (totp_records[i].secret == NULL) {
+            num_totp_records = i - 1;
+            // We can't easily undo at this point, because we destroyed
+            // the offset info (secret is in a union). So let's just junk
+            // the remaining records.
+            break;
+        }
+        memcpy(totp_records[i].secret, current_secret, totp_records[i].secret_size);
+    }
```

Comment:
How does this impact memory usage? Copying every secret into memory has caused resets in the past, and the fix was to read them into memory one at a time.

Reply:
Yes - there is a warning about this in the doco (i.e. that using secure mode is more likely to make you run out of memory, and whether it works well is a bit config dependent).

Personally, I've run 10 codes on my device without issue, but it obviously depends what other faces are loaded.